src/main/java/com/project/ita5/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,65 @@
+package com.project.ita5.security;
+
+import com.auth0.jwt.JWT;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.project.ita5.user.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+import static com.auth0.jwt.algorithms.Algorithm.HMAC512;
+import static com.project.ita5.security.SecurityConstants.EXPIRATION_TIME;
+import static com.project.ita5.security.SecurityConstants.HEADER_STRING;
+import static com.project.ita5.security.SecurityConstants.SECRET;
+import static com.project.ita5.security.SecurityConstants.TOKEN_PREFIX;
+
+public class JwtAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+    private AuthenticationManager authenticationManager;
+
+    @Autowired //nzn
```

Comment:
Don't be like "Sodra", delete such comments.

Reply:
Not "Sodra" anymore